asmcomp/compilenv.ml

```diff
@@ -443,6 +451,13 @@ let report_error ppf = function
       fprintf ppf "%a@ contains the description for unit\
                    @ %s when %s was expected"
         Location.print_filename filename name modname
+  | Mismatching_for_pack(filename, pack_1, None) ->
+      fprintf ppf "%a@ was built with -for-pack %s, but this module isn't"
```

Comment:
Is "this module" the correct terminology?  I'm wondering if it should be more explicit that it corresponds to the current compilation unit.
Also, I would write "is not" instead of "isn't" here.